fleetctl/fleetctl.go

```diff
@@ -482,6 +482,36 @@ func getChecker() *ssh.HostKeyChecker {
 	return ssh.NewHostKeyChecker(keyFile)
 }
 
+func getUnitFile(file string) (*unit.UnitFile, error) {
+	var uf *unit.UnitFile
+	name := unitNameMangle(file)
+
+	log.Debugf("Looking up for Unit(%s) or its corresponding template", name)
```

Comment:
Ok, so yes we log in both cases since we have two different paths https://github.com/coreos/fleet/pull/1433/files#diff-197bb752bf0f52567eb6b0023ea65e40R550 if the template lookup from the registry succeed we log it otherwise if we fail and take the second path, we end up with the lookup from disk in getUnitFromFile() which will log for us. That's why I had to log inside getUnitFileFromTemplate(), it also falls back to getUnitFromFile() in case the template is not in the registry.
Now why I enclosed the whole logic with more debug messages like this one https://github.com/coreos/fleet/pull/1433/files#diff-197bb752bf0f52567eb6b0023ea65e40R511 the reason is in getUnitFromFile(). That function is used in other places, I don't want to update it and it logs before even succeeding here: https://github.com/coreos/fleet/blob/master/fleetctl/fleetctl.go#L494 but it may fail just in line 496 in NewUnitFile() during deserialize or any other error which is good since we catch both cases: the unit was found in disk but we failed to instantiate it, so we didn't reach this code path at all: https://github.com/coreos/fleet/pull/1433/files#diff-197bb752bf0f52567eb6b0023ea65e40R511

IMHO all these debug messages will be really helpful! thanks you!